### ISSUE TITLE
ComplexConvTranspose

### ISSUE THREAD

**Issue by H320** [wavefrontshaping/complexPyTorch#26]
[https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Convolution.cpp#L812](url)

`from typing import List, Optional

import torch
import torch.nn.functional as F
from torch import Tensor, nn

class ComplexConvTranspose1d(nn.ConvTranspose1d):

    def forward(self, input: Tensor, output_size: Optional[List[int]] = None) -> Tensor:
        if self.padding_mode != 'zeros':
            raise ValueError('Only `zeros` padding mode is supported for ConvTranspose1d')

        assert isinstance(self.padding, tuple)
        # One cannot replace List by Tuple or Sequence in "_output_padding" because
        # TorchScript does not support `Sequence[T]` or `Tuple[T, ...]`.
        num_spatial_dims = 1
        output_padding = self._output_padding(
            input, output_size, self.stride, self.padding, self.kernel_size,  # type: ignore[arg-type]
            num_spatial_dims, self.dilation)  # type: ignore[arg-type]

        i = torch.view_as_real(input)
        i_r = i[..., 0]
        i_i = i[..., 1]
        w = torch.view_as_real(self.weight)
        w_r = w[..., 0]
        w_i = w[..., 1]
        b = torch.view_as_real(self.bias)
        b_r = b[..., 0]
        b_i = b[..., 1]

        a = F.conv_transpose1d(i_r, w_r, b_r, self.stride, self.padding, output_padding, self.groups, self.dilation)
        b = F.conv_transpose1d(i_i, w_i, None, self.stride, self.padding, output_padding, self.groups, self.dilation)
        c = F.conv_transpose1d(i_r + i_i, w_r + w_i, b_r + b_i, self.stride, self.padding, output_padding, self.groups, self.dilation)

        out = torch.stack([a - b, c - a - b], dim=-1)
        return torch.view_as_complex(out)

class ComplexConvTranspose2d(nn.ConvTranspose2d):

    def forward(self, input: Tensor, output_size: Optional[List[int]] = None) -> Tensor:
        if self.padding_mode != 'zeros':
            raise ValueError('Only `zeros` padding mode is supported for ConvTranspose2d')

        assert isinstance(self.padding, tuple)
        # One cannot replace List by Tuple or Sequence in "_output_padding" because
        # TorchScript does not support `Sequence[T]` or `Tuple[T, ...]`.
        num_spatial_dims = 2
        output_padding = self._output_padding(
            input, output_size, self.stride, self.padding, self.kernel_size,  # type: ignore[arg-type]
            num_spatial_dims, self.dilation)  # type: ignore[arg-type]

        i = torch.view_as_real(input)
        i_r = i[..., 0]
        i_i = i[..., 1]
        w = torch.view_as_real(self.weight)
        w_r = w[..., 0]
        w_i = w[..., 1]
        b = torch.view_as_real(self.bias)
        b_r = b[..., 0]
        b_i = b[..., 1]

        a = F.conv_transpose2d(i_r, w_r, b_r, self.stride, self.padding, output_padding, self.groups, self.dilation)
        b = F.conv_transpose2d(i_i, w_i, None, self.stride, self.padding, output_padding, self.groups, self.dilation)
        c = F.conv_transpose2d(i_r + i_i, w_r + w_i, b_r + b_i, self.stride, self.padding, output_padding, self.groups, self.dilation)

        out = torch.stack([a - b, c - a - b], dim=-1)
        return torch.view_as_complex(out)

class ComplexConvTranspose3d(nn.ConvTranspose3d):

    def forward(self, input: Tensor, output_size: Optional[List[int]] = None) -> Tensor:
        if self.padding_mode != 'zeros':
            raise ValueError('Only `zeros` padding mode is supported for ConvTranspose3d')

        assert isinstance(self.padding, tuple)
        # One cannot replace List by Tuple or Sequence in "_output_padding" because
        # TorchScript does not support `Sequence[T]` or `Tuple[T, ...]`.
        num_spatial_dims = 3
        output_padding = self._output_padding(
            input, output_size, self.stride, self.padding, self.kernel_size,  # type: ignore[arg-type]
            num_spatial_dims, self.dilation)  # type: ignore[arg-type]

        i = torch.view_as_real(input)
        i_r = i[..., 0]
        i_i = i[..., 1]
        w = torch.view_as_real(self.weight)
        w_r = w[..., 0]
        w_i = w[..., 1]
        b = torch.view_as_real(self.bias)
        b_r = b[..., 0]
        b_i = b[..., 1]

        a = F.conv_transpose3d(i_r, w_r, b_r, self.stride, self.padding, output_padding, self.groups, self.dilation)
        b = F.conv_transpose3d(i_i, w_i, None, self.stride, self.padding, output_padding, self.groups, self.dilation)
        c = F.conv_transpose3d(i_r + i_i, w_r + w_i, b_r + b_i, self.stride, self.padding, output_padding, self.groups, self.dilation)

        out = torch.stack([a - b, c - a - b], dim=-1)
        return torch.view_as_complex(out)
`
